### ISSUE TITLE
fetching nodes times out on large files

### ISSUE THREAD

**Issue by joperron** [gatsbyjs/gatsby-source-wordpress-experimental#26]
Hello @TylerBarnes! I'm trying to use this new source plugin (looks amazing!) - but it seems to hang on video media files. I'm not sure if it's my server, but is there a way I can ignore them since they won't be processed by gatsby like the images are?

**Comment by TylerBarnes** [comment]
@joperron if it's trying to process them now I'll need to update the source plugin internally, nothing you can change to make that work right now. I'll try to reproduce today and fix it but if you have a reproduction /graphql API I can use that would be helpful!

**Comment by joperron** [comment]
ayyy! thanks! try this: http://newtrax-wp.1024apps.com/graphql

**Comment by TylerBarnes** [comment]
Ahh, it's actually because you have huge video files and it times out downloading them. I'm investigating some solutions and/or workarounds for this now

**Comment by joperron** [comment]
I figured that might be it - just deleted the videos to give it another run so if you see it working it's not a sudden miracle!

**Comment by TylerBarnes** [comment]
@joperron I'm working on a release that adds a way to prevent fetching files by mime type.

I'm thinking of adding something like this to plugin options:

```js
{
  options: {
    type: {
      MediaItem: {
	      localFile: {
	          excludeByMimeTypes: [`video/mp4`],
          },
      },
    }
  }
}
```

**Comment by TylerBarnes** [comment]
Once we can query the file size, I'll have a default max file size that's configurable

**Comment by joperron** [comment]
Looks good to me, that's what I was imagining as a stopgap, and the `fileSize` of course would be amazing!

**Comment by TylerBarnes** [comment]
@joperron upgrade to `gatsby-source-wordpress-experimental@0.7.12` for the `excludeByMimeTypes` option. Thanks for opening this issue! This is really good to know about 👍 I'll comment here once fileSize is added as well.

**Comment by joperron** [comment]
Thank you Tyler!

**Comment by TylerBarnes** [comment]
FYI this fileSize option has been merged into WPGraphQL and we're just waiting on a new release 👍

**Comment by joperron** [comment]
That's huge! No pun intended

**Comment by TylerBarnes** [comment]
The `options.type.MediaItem.localFile.maxFileSizeBytes` option was added a couple weeks ago and is set to by default exclude media item files that are over 15Mb. https://github.com/gatsbyjs/gatsby-source-wordpress-experimental/blob/master/docs/plugin-options.md#typemediaitemlocalfilemaxfilesizebytes-number

Update to the latest version of the plugin to use it :)
